src/pymodaq/daq_viewer/daq_viewer_main.py

```diff
@@ -60,12 +60,76 @@
 
 
 class DAQ_Viewer(QObject):
-    """
+    """Main class of the DAQ_Viewer module.
+
+    It defines the UI of the module and ensure the communication between the UI and all the other objects that are
+    needed to run the module.
+
+    Signals
+    -------
+    grab_done_signal : pyqt signal
+        This signal is emitted each time an acquisition from the detector is ready. It contains the acquisition buffer.
+        It is emitted by show_data and get_data_from_viewer methods.
+        It is connected to the save_export_data method, and may be connected (depending on the context defined by the
+        user) to other modules that need the detector acquisition (e.g. Logger extension module to save the
+        acquisition).
+    command_detector : pyqt signal
+        This signal allows sending commands from the DAQ_Viewer thread to the DAQ_Detector thread.
+        It is connected to DAQ_Detector.queue_command. This last method will call the proper method of the DAQ_Detector
+        depending on the string that is sent in the ThreadCommand.
+
+    Parameters
+    ----------
+    refresh_time : int
+        Time delay (in ms) between two updates of the viewer in the UI. This time delay (contrary to
+        DAQ_Viewer.wait_time) will not directly affect the actual speed of the acquisition. If for example the
+        acquisition loop runs at 50 Hz, there is no point to display on the UI at such a speed. In that case the user
+        can specify a refresh time longer than 20 ms. Notice that the longer the refresh time, the better the
+        performances, since the displaying is quite costly in terms of resources. A very short refresh time can slow
+        down the program. For most applications, a refreshing time higher than 200 ms is enough. If the user wants to
+        optimize the performance of the program, he can untick the "Show data and process" parameter on the UI. If the
+        user chooses a refreshing time shorter than the delay between two acquisitions, then it just means that the
+        viewer will display all the acquisitions. Notice also that the ROI lineouts are calculated each time the
+        viewer is refreshed. It means that (up to now) the user cannot perform ROI lineouts calculation and saving
+        faster than the refresh time of the viewer. This is probably not an expected behavior.
```

Comment:
You are welcome! :) I needed to write it down :P

What I meant for the unexpected behavior is that for example if I want to save some ROI lineouts from a viewer with the Logger extension, I cannot turn off the visualization of the data. The behavior of the saving depends on the visualization on the UI. Maybe this can be easily corrected by modifying the action when the user uncheck "Show data and process".

Reply:
if you want efficient you don't want to display everything and moreover you may want to save only raw data. So I guess if you want to log ROI you can stick with "show data"